cloudplatform/connectivity-oauth/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/MultiUrlPropertySupplier.java

```diff
@@ -114,7 +139,18 @@ static final class Builder<T extends OptionsEnhancer<T>>
         @Nonnull
         Builder<T> withUrlKey( @Nonnull final OptionsEnhancer<T> enhancer, @Nonnull final String urlKey )
         {
-            urlKeys.put(enhancer, urlKey);
+            return withUrlKey(enhancer, urlKey, () -> enhancerClass == BtpServiceOptions.BusinessLoggingOptions.class);
+        }
+
+        @Nonnull
+        private Builder<T> withUrlKey(
+            @Nonnull final OptionsEnhancer<T> enhancer,
+            @Nonnull final String urlKey,
+            @Nonnull final Supplier<Boolean> isPathless )
```

Comment:
discussed offline a bit, leaving this here before I forget it again: another generic solution alternative might be:

```suggestion
            @Nonnull final Function<URI, URI> uriTransformation )
```

Reply:
Would this not mean then that we'd be computing the URIs on builder init rather than when retrieving via `getServiceURI()`. I'm not sure of the typical usage patterns here, if we tend to `get` the URIs soon after construction then it shouldn't be an issue, but I would assume since the original logic is in the get method that we prefer the lazy approach?

Reply:
The function would be invoked at the same point in time where currently `Supplier<Boolean>` is invoked, so there would not be any difference in that I think..

For most cases the function would be identity, but for the case where we want to modify the URI it could be e.g. `uri -> uri.resolve("/")`. The advantage over `Supplier<Boolean>` would be that you can do anything to the URL, e.g. appending a path or some query even maybe..

Reply:
Ah okay, I misunderstood how a `Function<>` was used then. Thanks for the clarification 👍